In [20]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [21]:
train_path = r"C:\Users\kaust\Desktop\PG\SEM_2\DA Project\archive\train.csv"
train_data = pd.read_csv(train_path)
test_path = r"C:\Users\kaust\Desktop\PG\SEM_2\DA Project\archive\test.csv"
test_data = pd.read_csv(test_path)

In [22]:
train_data.head()

UniqueID  disbursed_amount  asset_cost    ltv  branch_id  supplier_id  \
0    420825             50578       58400  89.55         67        22807   
1    537409             47145       65550  73.23         67        22807   
2    417566             53278       61360  89.63         67        22807   
3    624493             57513       66113  88.48         67        22807   
4    539055             52378       60300  88.39         67        22807   

   manufacturer_id  Current_pincode_ID Date.of.Birth Employment.Type  ...  \
0               45                1441      01-01-84        Salaried  ...   
1               45                1502      31-07-85   Self employed  ...   
2               45                1497      24-08-85   Self employed  ...   
3               45                1501      30-12-93   Self employed  ...   
4               45                1495      09-12-77   Self employed  ...   

  SEC.SANCTIONED.AMOUNT  SEC.DISBURSED.AMOUNT  PRIMARY.INSTAL.AMT  \
0                     0                     0                   0   
1                     0                     0                1991   
2                     0                     0                   0   
3                     0                     0                  31   
4                     0                     0                   0   

   SEC.INSTAL.AMT  NEW.ACCTS.IN.LAST.SIX.MONTHS  \
0               0                             0   
1               0                             0   
2               0                             0   
3               0                             0   
4               0                             0   

   DELINQUENT.ACCTS.IN.LAST.SIX.MONTHS  AVERAGE.ACCT.AGE  \
0                                    0         0yrs 0mon   
1                                    1        1yrs 11mon   
2                                    0         0yrs 0mon   
3                                    0         0yrs 8mon   
4                                    0         0yrs 0mon   

   CREDIT.HISTORY.LENGTH  NO.OF_INQUIRIES  loan_default  
0              0yrs 0mon                0             0  
1             1yrs 11mon                0             1  
2              0yrs 0mon                0             0  
3              1yrs 3mon                1             1  
4              0yrs 0mon                1             1  

[5 rows x 41 columns]

In [23]:
train_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 233154 entries, 0 to 233153
Data columns (total 41 columns):
 #   Column                               Non-Null Count   Dtype  
---  ------                               --------------   -----  
 0   UniqueID                             233154 non-null  int64  
 1   disbursed_amount                     233154 non-null  int64  
 2   asset_cost                           233154 non-null  int64  
 3   ltv                                  233154 non-null  float64
 4   branch_id                            233154 non-null  int64  
 5   supplier_id                          233154 non-null  int64  
 6   manufacturer_id                      233154 non-null  int64  
 7   Current_pincode_ID                   233154 non-null  int64  
 8   Date.of.Birth                        233154 non-null  object 
 9   Employment.Type                      225493 non-null  object 
 10  DisbursalDate                        233154 non-null  object 
 11  State_ID     

In [24]:
unique_counts = train_data.nunique()
print(unique_counts)

UniqueID                               233154
disbursed_amount                        24565
asset_cost                              46252
ltv                                      6579
branch_id                                  82
supplier_id                              2953
manufacturer_id                            11
Current_pincode_ID                       6698
Date.of.Birth                           15433
Employment.Type                             2
DisbursalDate                              84
State_ID                                   22
Employee_code_ID                         3270
MobileNo_Avl_Flag                           1
Aadhar_flag                                 2
PAN_flag                                    2
VoterID_flag                                2
Driving_flag                                2
Passport_flag                               2
PERFORM_CNS.SCORE                         573
PERFORM_CNS.SCORE.DESCRIPTION              20
PRI.NO.OF.ACCTS                   

# Since UniqueID is useless its better to drop it 

In [25]:
train_data.drop(columns=['UniqueID'], inplace=True)  
test_data.drop(columns=['UniqueID'], inplace=True)  

In [28]:
train_data.describe().T

count           mean           std  \
disbursed_amount                     233154.0   54356.993528  1.297131e+04   
asset_cost                           233154.0   75865.068144  1.894478e+04   
ltv                                  233154.0      74.746530  1.145664e+01   
branch_id                            233154.0      72.936094  6.983499e+01   
supplier_id                          233154.0   19638.635035  3.491950e+03   
manufacturer_id                      233154.0      69.028054  2.214130e+01   
Current_pincode_ID                   233154.0    3396.880247  2.238148e+03   
State_ID                             233154.0       7.262243  4.482230e+00   
Employee_code_ID                     233154.0    1549.477148  9.752613e+02   
MobileNo_Avl_Flag                    233154.0       1.000000  0.000000e+00   
Aadhar_flag                          233154.0       0.840320  3.663097e-01   
PAN_flag                             233154.0       0.075577  2.643201e-01   
VoterID_flag                         233154.0       0.144943  3.520439e-01   
Driving_flag                         233154.0       0.023242  1.506720e-01   
Passport_flag                        233154.0       0.002127  4.607421e-02   
PERFORM_CNS.SCORE                    233154.0     289.462994  3.383748e+02   
PRI.NO.OF.ACCTS                      233154.0       2.440636  5.217233e+00   
PRI.ACTIVE.ACCTS                     233154.0       1.039896  1.941496e+00   
PRI.OVERDUE.ACCTS                    233154.0       0.156549  5.487867e-01   
PRI.CURRENT.BALANCE                  233154.0  165900.076936  9.422736e+05   
PRI.SANCTIONED.AMOUNT                233154.0  218503.855323  2.374794e+06   
PRI.DISBURSED.AMOUNT                 233154.0  218065.898655  2.377744e+06   
SEC.NO.OF.ACCTS                      233154.0       0.059081  6.267946e-01   
SEC.ACTIVE.ACCTS                     233154.0       0.027703  3.160566e-01   
SEC.OVERDUE.ACCTS                    233154.0       0.007244  1.110789e-01   
SEC.CURRENT.BALANCE                  233154.0    5427.792819  1.702370e+05   
SEC.SANCTIONED.AMOUNT                233154.0    7295.923347  1.831560e+05   
SEC.DISBURSED.AMOUNT                 233154.0    7179.997873  1.825925e+05   
PRIMARY.INSTAL.AMT                   233154.0   13105.481720  1.513679e+05   
SEC.INSTAL.AMT                       233154.0     323.268449  1.555369e+04   
NEW.ACCTS.IN.LAST.SIX.MONTHS         233154.0       0.381833  9.551067e-01   
DELINQUENT.ACCTS.IN.LAST.SIX.MONTHS  233154.0       0.097481  3.844390e-01   
NO.OF_INQUIRIES                      233154.0       0.206615  7.064977e-01   
loan_default                         233154.0       0.217071  4.122523e-01   

                                            min       25%      50%       75%  \
disbursed_amount                       13320.00  47145.00  53803.0  60413.00   
asset_cost                             37000.00  65717.00  70946.0  79201.75   
ltv                                       10.03     68.88     76.8     83.67   
branch_id                                  1.00     14.00     61.0    130.00   
supplier_id                            10524.00  16535.00  20333.0  23000.00   
manufacturer_id                           45.00     48.00     86.0     86.00   
Current_pincode_ID                         1.00   1511.00   2970.0   5677.00   
State_ID                                   1.00      4.00      6.0     10.00   
Employee_code_ID                           1.00    713.00   1451.0   2362.00   
MobileNo_Avl_Flag                          1.00      1.00      1.0      1.00   
Aadhar_flag                                0.00      1.00      1.0      1.00   
PAN_flag                                   0.00      0.00      0.0      0.00   
VoterID_flag                               0.00      0.00      0.0      0.00   
Driving_flag                               0.00      0.00      0.0      0.00   
Passport_flag                              0.00      0.00      0.0      0.00   
PERFORM_CNS.SCORE            

In [30]:
train_data.isnull().sum()

disbursed_amount                          0
asset_cost                                0
ltv                                       0
branch_id                                 0
supplier_id                               0
manufacturer_id                           0
Current_pincode_ID                        0
Date.of.Birth                             0
Employment.Type                        7661
DisbursalDate                             0
State_ID                                  0
Employee_code_ID                          0
MobileNo_Avl_Flag                         0
Aadhar_flag                               0
PAN_flag                                  0
VoterID_flag                              0
Driving_flag                              0
Passport_flag                             0
PERFORM_CNS.SCORE                         0
PERFORM_CNS.SCORE.DESCRIPTION             0
PRI.NO.OF.ACCTS                           0
PRI.ACTIVE.ACCTS                          0
PRI.OVERDUE.ACCTS               

## There are 7661 null values in the Employment Type attribute so we need to fix that by either imputation or dropping them

In [32]:
print(train_data['Employment.Type'])

0              Salaried
1         Self employed
2         Self employed
3         Self employed
4         Self employed
              ...      
233149         Salaried
233150    Self employed
233151         Salaried
233152         Salaried
233153         Salaried
Name: Employment.Type, Length: 233154, dtype: object
